service_container/service_subscribers_locators.rst

```diff
@@ -110,18 +110,30 @@ in the service subscriber::
     that you have :ref:`autoconfigure <services-autoconfigure>` enabled. You
     can also manually add the ``container.service_subscriber`` tag.
 
-The injected service is an instance of :class:`Symfony\\Component\\DependencyInjection\\ServiceLocator`
-which implements both the PSR-11 ``ContainerInterface`` and :class:`Symfony\\Contracts\\Service\\ServiceProviderInterface`.
-It is also a callable and a countable::
+The injected service is an instance of
+:class:`Symfony\\Component\\DependencyInjection\\ServiceLocator`. A service
```

Comment:
I still wouldn't mention this, it's an implementation detail.
We should tell about ServiceCollectionInterface: if ppl type-hint for it, *then* they will get an object that is coutable+iterable, in addition to being a PSR11 locator.